flang/test/Lower/transformational-intrinsics.f90

```diff
@@ -124,4 +124,18 @@ subroutine in_elem_expr(x, y, z)
   ! CHECK: fir.freemem %[[res_addr]] : !fir.heap<!fir.array<?x!fir.logical<1>>>
 end subroutine
 
+! UNPACK
+! CHECK-LABEL: unpack_test
+subroutine unpack_test()
+  integer, dimension(3) :: vector
+  integer, dimension (3,3) :: field
+
+  logical, dimension(3,3) :: mask
+  integer, dimension(3,3) :: result
+  result = unpack(vector, mask, field)
+  ! CHECK: fir.call @_FortranAUnpack(%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}) : (!fir.ref<!fir.box<none>>, !fir.box<none>, !fir.box<none>, !fir.box<none>, !fir.ref<i8>, i32) -> none
+  result = unpack(vector, mask, 343)
+  ! CHECK: fir.call @_FortranAUnpack(%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}) : (!fir.ref<!fir.box<none>>, !fir.box<none>, !fir.box<none>, !fir.box<none>, !fir.ref<i8>, i32) -> none
```

Comment:
Shouldn't these do at least some cursory CHECKs on the arguments?

Reply:
Take a look at how some of the other tests are done, like reshape. As Eric said, you should check the arguments. Also, add a check on the call to fir.freemem at the end.

Reply:
I wasn't sure about the level of checking needed for argument passing.  The other tests don't seem consistent about this.  I'll add some checks.

Reply:
I think we'll want to write more restrictive tests before we try to upstream to help prevent helpful rerouting suggestions sending us over cliffs.

Reply:
I'll add checks for the arguments.